src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -82,122 +90,152 @@ private void TryToDeleteCachedImages()
 			}
 		}
 
-		public string GetPathToResizedImage(string originalPath)
+		public string GetPathToResizedImage(string originalPath, bool getThumbnail = false)
 		{
 			//don't mess with Bloom UI images
 			if (new[] {"/img/", "placeHolder", "Button"}.Any(s => originalPath.Contains(s)))
 				return originalPath;
 
+			// check if this image is in the do-not-process list
+			bool test;
+			if (_imageFilesToReturnUnprocessed.TryGetValue(originalPath, out test)) return originalPath;
```

Comment:
Surely this needs to use the cacheFileName as key? As it stands, we will never make a thumbnail of a jpg, once MakePngBackgroundTransparent() returns false for it; nor will we ever fix the background of an image small enough not to need a thumbnail, if we happen to try that first.